module_utils/network_lsr/argument_validator.py

```diff
@@ -413,6 +414,73 @@ def _validate_post(self, value, name, result):
 
         return result
 
+class ArgValidatorIPRule(ArgValidatorDict):
+    def __init__(self, name, family=None, required=False, default_value=None):
+        ArgValidatorDict.__init__(
+            self,
+            name,
+            required,
+            nested=[
+                ArgValidatorIP(
+                    "from", family=family, required=False, plain_address=False
+                ),
+                ArgValidatorNum("from_prefix", default_value=32, val_min=0, val_max=32),
+                ArgValidatorIP(
+                    "to", family=family, required=False, plain_address=False
+                ),
+                ArgValidatorNum("to_prefix", default_value=32, val_min=0, val_max=32),
+                ArgValidatorStr(
+                    "iif", default_value=None, enum_values=listdir('/sys/class/net/')
+                ),
+                ArgValidatorStr(
+                    "oif", default_value=None, enum_values=listdir('/sys/class/net/')
+                ),
+                ArgValidatorNum(
+                    "tos", default_value=None, val_min=0, val_max=255
+                ),
+                ArgValidatorNum(
+                    "dsfield", default_value=None, val_min=0, val_max=255
+                ),
+                ArgValidatorNum(
+                    "fwmark", default_value=None, val_min=0, val_max=65535
+                ),
+                # Not implemented
+                # ArgValidatorRange(
+                #     "uidrange", default_value=None, val_min=0, val_max=65535
+                # ),
+                ArgValidatorNum(
+                    "ipproto", default_value=None, val_min=0, val_max=255
+                ),
+                # ArgValidatorRange(
+                #     "sport", default_value=None, val_min=0, val_max=65535
+                # ),
+                # ArgValidatorDport(
+                #     "dport", default_value=None, val_min=0, val_max=65535
+                # ),
+                ArgValidatorNum(
+                    "priority", default_value=None, val_min=0, val_max=32767
+                ),
+                ArgValidatorStr(
+                    "table", default_value="main"
```

Comment:
I think the table should first an foremost be a uint32. That is certainly what `ArgValidatorIPRule` should expose after parsing. Possibly the property can accept aliases like "main" or "local", but then this should be validated and parsed to an integer by `ArgValidatorIPRule`.

Reply:
The problem here is, that you can specify `table` with a name. I did not yet implement that, because I had no idea whether the module or some task in the role should take care of this. Good place to answer :)

Reply:
iproute2 allows you to specify the table by name. That will look into `/etc/iproute2/rt_tables`. I think that should explicitly *not* be supported. I would only support specifying the table by number.

If you care, we can allow certain aliases (`main` ==  254, `local` == 255, `default` == 253), but that should *not* look at `/etc/iproute2/rt_tables`, because then the behaviour of the role depends on some files on that target system.

Reply:
TBH I don't care in my boring use case of having 3 different route tables, but why do you think the role should not do it? We can add a task before the module is called and ensure all named tables exist in `/etc/iproute2/rt_tables`. Why do you think the module can't rely on a prerequisite, that is satisfied by the role which calls the module? I would understand such approach if that module would have ambitions to become part of Ansible core.

Reply:
* it's extra work
* because then the behaviour of the playbook depends on files in /etc. I think that is generally undesirable.

Reply:
I don't think we are on the same wave, but let's keep it numbered for now and possibly add names later.